### PR TITLE
Change Network to derive Deserialize

### PR DESCRIPTION
**Pull request by bbalser**

### CODE REVIEW

**Review by jeffgrunewald — APPROVED** [review]

File: src/lib.rs [hunk 1]
```diff
@@ -239,3 +241,59 @@ pub trait ReadFrom {
...
+            &[Token::UnitVariant {
+                name: "KeyType",
+                variant: "ed25519",
+            }],
+        );
+        assert_de_tokens(
+            &KeyType::EccCompact,
+            &[Token::UnitVariant {
+                name: "KeyType",
+                variant: "ecccompact",
```
Comment (madninja) [inline src/lib.rs:288]: this ain't right the string representation for EccCompact is `ecc_compact`.. as encoded by `FromStr`.. You'l need a rename for that specific variant

**Review by madninja — CHANGES_REQUESTED** [review]

File: src/lib.rs (hunk 1 above)
Comment (bbalser) [reply src/lib.rs]: Fixed fd66457

**Review by madninja — APPROVED** [review]

File: src/lib.rs [hunk 2]
```diff
@@ -239,3 +242,59 @@ pub trait ReadFrom {
...
+            &[Token::UnitVariant {
+                name: "Network",
+                variant: "other",
+            }],
+            "unknown variant `other`, expected `mainnet` or `testnet`",
+        );
+    }
+
+    #[test]
+    fn keytype_can_deserialize() {
```
Comment (JayKickliter) [inline src/lib.rs:277]: The clippy errors have been fixed, now revealing that this test fails.

**Review by JayKickliter — CHANGES_REQUESTED** [review]

File: src/lib.rs (hunk 2 above)
Comment (jeffgrunewald) [reply src/lib.rs:277]: How is this test run? `cargo test` doesn’t fail at all for me

Comment (jeffgrunewald) [reply src/lib.rs:277]: Nvm, I see it’s running features-specific tests with a flag

**Review by JayKickliter — APPROVED** [review]
Merge at your will (if you're allowed to)

**Pull request merged**